src/transformers/retrieval_rag.py

```diff
@@ -190,53 +190,76 @@ def get_top_docs(self, question_hidden_states: np.ndarray, n_docs=5) -> Tuple[np
         return np.array(ids), np.array(vectors)
 
 
-class HFIndex:
+class HFIndexBase:
```

Comment:
That's slightly backwards breaking, but OK IMO since people probably didn't use the class directly